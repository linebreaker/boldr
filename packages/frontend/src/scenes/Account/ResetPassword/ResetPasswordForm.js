/* @flow */

import React from 'react';
import { Field, reduxForm } from 'redux-form';
import Button from '@boldr/ui/Button';
import Form, { TextFormField } from '@boldr/ui/Form';

const style = {
  margin: 12,
};
export type Props = {
  handleSubmit: Function,
  submitting?: boolean,
  pristine?: boolean,
};
const ResetPasswordForm = (props: Props) => {
  const { handleSubmit, pristine, submitting } = props;
  return (
    <Form onSubmit={handleSubmit} className="boldrui-form boldrui-form--reset">
      <Field
        id="password"
        name="password"
        type="password"
        label="Enter a new password"
        component={TextFormField}
      />
      <Field
        id="confirm"
        name="confirm"
        type="password"
        label="Confirm new password."
        component={TextFormField}
      />
      <Button htmlType="submit" style={style} kind="primary" disabled={submitting || pristine}>
        Reset Password
      </Button>
    </Form>
  );
};

export default reduxForm({
  form: 'resetPasswordForm',
})(ResetPasswordForm);
