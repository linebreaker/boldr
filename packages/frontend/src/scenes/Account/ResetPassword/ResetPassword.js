/* @flow */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import FormCard from '@boldr/ui/Form/FormCard';
import { resetPassword } from '../state/actions';
import ResetPasswordForm from './ResetPasswordForm';

export type Props = {
  dispatch: () => void,
  match: Object,
};

class ResetPassword extends Component {
  constructor(props: Props) {
    super(props);

    (this: any).handleReset = this.handleReset.bind(this);
  }
  props: Props;

  handleReset(values: Object): void {
    this.props.dispatch(resetPassword(values.password, this.props.match.params.token));
  }

  render() {
    return (
      <div className="resetpw-wrapper">
        <Helmet title="Reset Password" />
        <FormCard
          title="Reset Password"
          skinny
          lightText
          form={<ResetPasswordForm onSubmit={this.handleReset} />}
        />
      </div>
    );
  }
}

export default connect()(ResetPassword);
