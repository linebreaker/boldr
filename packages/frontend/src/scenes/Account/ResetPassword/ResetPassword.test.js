import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import ResetPassword from './ResetPassword';

test('<ResetPassword />, renders the reset password form card', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {};
  const store = mockStore(initialState);
  const wrapper = shallow(
    <Provider store={store}>
      <ResetPassword />
    </Provider>,
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
