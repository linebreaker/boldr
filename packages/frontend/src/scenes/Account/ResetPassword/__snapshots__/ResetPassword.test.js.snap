// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassword />, renders the reset password form card 1`] = `<Connect(ResetPassword) />`;
